core/src/main/java/fj/function/Booleans.java

```diff
@@ -92,6 +84,56 @@ public static boolean and(final List<Boolean> l) {
     return Monoid.conjunctionMonoid.sumLeft(l);
   }
 
+    /**
+     * maps given function to the predicate function
+     * @param p predicate to be mapped over
+     * @param f function
+     * @return predicate function
+     */
+    public static <A, B>  F<B, Boolean> contramap(F<B, A> f, F<A, Boolean> p){
+        return compose(p, f);
+    }
+
+    /**
+     * returns inverse of contramap
+     * @param p predicate to be mapped over
+     * @param f function
+     * @return predicate function
+     */
+    public static <A, B>  F<B, Boolean> isnot(F<B, A> f, F<A, Boolean> p){
+        return compose(not, contramap(f, p));
+    }
+
```

Comment:
IHMO, `contramap` is more appropriate as it is in Data.Covariant.Functor http://hackage.haskell.org/package/contravariant-0.1.2/docs/src/Data-Functor-Contravariant.html. In fact, I couldn't think of better name for `isnot` so I preferred to settle down as it was named in Data.Function.Predicate